docs/src/guide/tutorial.jl

```diff
@@ -0,0 +1,298 @@
+# an LDA model, which consists of the following terms[^2]:
+#
+# [^2]: If you are not familiar with Julia syntax, this is equivalent to `[typeof(t) for t in model.term_types]`.
+
+typeof.(model.term_types)
+
+# DFTK computes energies for all terms of the model individually,
+# which are available in `scfres.energies`:
+
+scfres.energies
+#
+# For now the following energy terms are known to DFTK:
+
+# - Kinetic energy
+# - Local potential energy, either given by analytic potentials or
+#   specified by the type of atoms.
+# - Nonlocal potential energy, for norm-conserving pseudopotentials
+# - Nuclei energies (eg Ewald or pseudopotential correction)
+# - Hartree energy
+# - Exchange-correlation energy
+# - Power nonlinearities (useful for Gross-Pitaevskii type models)
+# - Magnetic field energy
+# - Entropy term
+
+# Custom types can be added if needed. For examples see
+# the definition of the above terms in the
+# [`src/terms`](https://dftk.org/tree/master/src/terms) directory.
+#src TODO Refer to an example showing how to add / build custom terms
+
+# By mixing and matching these terms, the user can create custom models
+# not limited to DFT. Convenience constructors are provided for common cases:
+#
+# - `model_LDA`: LDA model using the
+#   [Teter parametrisation](https://doi.org/10.1103/PhysRevB.54.1703)
+# - `model_DFT`: Assemble a DFT model using
+#    any of the LDA or GGA functionals of the
+#    [libxc](https://tddft.org/programs/libxc/functionals/) library,
+#    for example:
+#    ```
+#    model_DFT(lattice, atoms, [:gga_x_pbe, :gga_c_pbe])
+#    model_DFT(lattice, atoms, :lda_xc_teter93)
+#    ```
+#    where the latter is equivalent to `model_LDA`.
+#    Specifying no functional is the reduced Hartree-Fock model:
+#    ```
+#    model_DFT(lattice, atoms, [])
+#    ```
+# - `model_atomic`: A linear model, which contains no electron-electron interaction
+#   (neither Hartree nor XC term).
+
+
+# ## `PlaneWaveBasis` and plane-wave discretisations
+#src TODO Some of this is a bit technical and should probably be factored
+#src      out into a "background theory section"
+
+# The `PlaneWaveBasis` datastructure handles the discretization of a
+# given `Model` in a plane-wave basis.
+# As usual in plane-wave methods the discretization is twofold:
+# Once the ``k``-point grid, which determines the sampling
+# *inside* the Brillouin zone and on top of that a finite
+# plane-wave grid to discretise the lattice-periodic functions.
+# The former aspect is controlled by the `kgrid` argument
+# of `PlaneWaveBasis`, the latter is controlled by the
+# cutoff energy parameter `Ecut`:
+
+PlaneWaveBasis(model, Ecut; kgrid=kgrid);
+
+# Another option to generate a `PlaneWaveBasis` is to pass an explicit list of
+# irreducible `k`-Points and associated symmetry operations to form
+# additional reducible `k`-Points from them, see
+#md # [Crystal symmetries](@ref).
+#nb # https://juliamolsim.github.io/DFTK.jl/dev/guide/symmetries.html.
+
```

Comment:
Too detailed. Just say "by default this uses symmetry to reduce the number of kpoints, see ref"